src/gsy_e/models/state.py

```diff
@@ -668,11 +695,18 @@ def add_default_values_to_state_profiles(self, future_time_slots):
                                    ESSEnergyOrigin.LOCAL: 0.,
                                    ESSEnergyOrigin.EXTERNAL: 0.})
 
-    def market_cycle(self, past_time_slot, current_time_slot: DateTime, all_future_time_slots):
+    def market_cycle(self, past_time_slot, current_time_slot: DateTime,
+                     all_future_time_slots: List[DateTime]):
         """
         Simulate actual Energy flow by removing pledged storage and adding bought energy to the
         used_storage
         """
+        if GlobalConfig.FUTURE_MARKET_DURATION_HOURS:
+            # In case the future market is enabled, the future orders have to be deleted once
+            # the market becomes a spot market
+            del self.offered_buy_kWh[current_time_slot]
```

Comment:
Thank you, that's perfect and exactly what I had in mind when we discussed this! I would prefer instead of deleting the key to have normal assignments to zero `self.offered_buy_kWh[current_time_slot] = 0.` , however it is a matter of personal preference therefore feel free to leave it as is.